probe_scraper/runner.py

```diff
@@ -233,11 +246,27 @@ def load_glean_metrics(cache_dir, out_dir, repositories_file, dry_run, glean_rep
                         "message": msg
                     })
 
+            try:
+                config = {'allow_reserved': repo_name == 'glean'}
+                if pings_files:
+                    results, errs = GLEAN_PINGS_PARSER.parse(pings_files, config)
+                    pings[repo_name][commit_hash] = results
+            except Exception:
+                msg = "Improper file in {}\n{}".format(', '.join(pings_files),
+                                                       traceback.format_exc())
+                emails[repo_name]["emails"].append({
+                    "subject": "Probe Scraper: Improper File",
+                    "message": msg
+                })
+
```

Comment:
This section very closely mirrors the above for metrics. Let's try and pull this out into shared functionality.

Reply:
We should also email on errors (as we do above).